docs(dual-region): add all related dual-region topics (#3562)
camunda/camunda-docs: docs/self-managed/operational-guides/multi-region/dual-region-ops.md

This means those records will not be included in the surviving region's Elasticsearch backup when the recreated region's Elasticsearch is restored from the backup, leading to the new region missing those records (as Zeebe does not re-export them).

:::

You are preparing everything for the newly created region to take over again to restore the functioning dual-region setup.

For this, stop the Zeebe exporters from exporting any new data to Elasticsearch so you can create an Elasticsearch backup.

Additionally, temporarily scale down Operate and Tasklist to zero replicas. This will result in users not being able to interact with Camunda 8 anymore and is required to guarantee no new data is imported to Elasticsearch.

:::note

This **does not** affect the processing of process instances in any way. The impact is that process information about the affected instances might not be visible in Operate and Tasklist.

:::

#### How to get there

1. Disable Operate and Tasklist by scaling to 0:

```bash

> kubectl --context $CLUSTER_SURVIVING scale deployments/$HELM_RELEASE_NAME-operate --replicas 0

Reviewer: ```suggestion
kubectl --context $CLUSTER_SURVIVING scale -n $ CAMUNDA_NAMESPACE_SURVIVING deployments/$HELM_RELEASE_NAME-operate --replicas 0
```

Author: nice catch!

> kubectl --context $CLUSTER_SURVIVING scale deployments/$HELM_RELEASE_NAME-tasklist --replicas 0

Reviewer: ```suggestion
kubectl --context $CLUSTER_SURVIVING -n $CAMUNDA_NAMESPACE_SURVIVING scale deployments/$HELM_RELEASE_NAME-tasklist --replicas 0
```

> 

Reviewer: ```suggestion
```

Reviewer: (trailing new line not needed)

:::info

The procedure works for other cloud providers and bare metal. You have to adjust the AWS S3-specific part depending on your chosen backup source for Elasticsearch. Consult the [Elasticsearch documentation on snapshot and restore](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshot-restore.html) to learn more about this, and specifically the [different supported types](https://www.elastic.co/guide/en/elasticsearch/reference/current/snapshots-register-repository.html#ess-repo-types) by Elasticsearch.

:::

1. Determine the S3 bucket name by retrieving it via Terraform. Go to `aws/dual-region/terraform` within the repository and retrieve the bucket name from the Terraform state:

```bash
export S3_BUCKET_NAME=$(terraform output -raw s3_bucket_name)
```

2. Configure Elasticsearch backup endpoint in the surviving namespace `CAMUNDA_NAMESPACE_SURVIVING`:

```bash
ELASTIC_POD=$(kubectl --context $CLUSTER_SURVIVING get pod --selector=app\.kubernetes\.io/name=elasticsearch -o jsonpath='{.items[0].metadata.name}' -n $CAMUNDA_NAMESPACE_SURVIVING)

> kubectl --context $CLUSTER_SURVIVING exec -n $CAMUNDA_NAMESPACE_SURVIVING -it $ELASTIC_POD -- curl -XPUT "http://localhost:9200/_snapshot/camunda_backup" -H "Content-Type: application/json" -d'

Reviewer: shall the container name be used here explicitly?
tried the command, and got this warning:
```
Defaulted container "elasticsearch" out of: elasticsearch, sysctl (init)
```

Reviewer: (same for other `kubectl exec commands`)

Reviewer: can also keep it as is, not a big deal

Author: don't have a strong preference since it will default to the 1st container, which typically is the main one.
Nevertheless, to be explicit added the `-c elasticsearch`.

  --version $HELM_CHART_VERSION \
  --kube-context $CLUSTER_SURVIVING \
  --namespace $CAMUNDA_NAMESPACE_SURVIVING \
  -f camunda-values.yml \
  -f $REGION_SURVIVING/camunda-values.yml \
  --set operate.enabled=false \
  --set tasklist.enabled=false
```

5. Upgrade the failover Camunda environment in `CAMUNDA_NAMESPACE_FAILOVER` and `REGION_SURVIVING` to point to the new Elasticsearch:

```bash
helm upgrade $HELM_RELEASE_NAME camunda/camunda-platform \
  --version $HELM_CHART_VERSION \
  --kube-context $CLUSTER_SURVIVING \
  --namespace $CAMUNDA_NAMESPACE_FAILOVER \
  -f camunda-values.yml \
  -f $REGION_SURVIVING/camunda-values-failover.yml
```

6. Upgrade the new region environment in `CAMUNDA_NAMESPACE_RECREATED` and `REGION_RECREATED` to point to the new Elasticsearch:

```bash
helm upgrade $HELM_RELEASE_NAME camunda/camunda-platform \
  --version $HELM_CHART_VERSION \
  --kube-context $CLUSTER_RECREATED \
  --namespace $CAMUNDA_NAMESPACE_RECREATED \
  -f camunda-values.yml \
  -f $REGION_RECREATED/camunda-values.yml \
  --set global.multiregion.installationType=failBack \
  --set operate.enabled=false \
  --set tasklist.enabled=false
```

> 7. Delete all the Zeebe broker pods in the new region `REGION_RECREATED`, as those are blocking a successful rollout of the config change due to the failback mode. The resulting recreated 4 Zeebe brokers pods are expected to be again in 2 functional and 2 sleeping pods due to the failback mode.

Reviewer: I'd suggest not harcoding the number of pods here:
```suggestion
7. Delete all the Zeebe broker pods in the new region `REGION_RECREATED`, as those are blocking a successful rollout of the config change due to the failback mode. The resulting recreated Zeebe brokers pods are expected to be again half of them being functional and half of them running in the sleeping mode due to the failback mode.
```
